controllers/argocd/util.go

```diff
@@ -1156,6 +1162,15 @@ type DeprecationEventEmissionStatus struct {
 // This is temporary and can be removed in v0.0.6 when we remove the deprecated fields.
 var DeprecationEventEmissionTracker = make(map[string]DeprecationEventEmissionStatus)
 
+// sourceNamespaceFilterPredicate returns a simple predicate allowing all create events.
+func sourceNamespaceFilterPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(ce event.CreateEvent) bool {
+			return true
+		},
+	}
+}
+
```

Comment:
what is the purpose of this predicate? I think the default is already `true` for any time of event that occurs on a watched resource, if a predicate is not specified

Not to mention, we already have a different `namespaceFilterPredicate` that returns false on all namespace update events that dont match an existing criterion involving the managed-by label

I'm wondering how these 2 predicates will interact with each other since they watch the same resource

Reply:
@jaideepr97 , I wanted to observe only create events within namespaces matching the glob pattern specified in the spec.sourceNamespace of ArgoCD and reconcile those ArgoCD instances that align with this pattern.

 Example : 
 Upon the creation of a namespace as test-2, and given an existing ArgoCD instance configured as follows:

apiVersion: argoproj.io/v1beta1
kind: ArgoCD
metadata:
  name: example-argocd
  namespace: default
spec:
  sourceNamespaces:
  - test-*
I  wanted to initiate the reconciliation process for this particular ArgoCD instance.

Reply:
If it is "true" by default  we can remove it

Reply:
@jaideepr97, I have removed the extra watch on the namespace resource and updated the namespaceResourceMapper to handle both ns with labels and ns without labels.